c7n/cli.py

```diff
@@ -132,6 +206,10 @@ def main():
     parser = setup_parser()
     options = parser.parse_args()
 
+    # policy-metrics requires some post-parsing validation
+    if options.subparser == 'policy-metrics':
```

Comment:
I did this first - when I moved over policymetrics.py.  Then when I started working on the schema subcommand I figured it was inappropriate to do the level of validation required for that in cli.py (e.g. cli.py should not know all the available resources).  I was aware of this incongruity and wasn't sure if we should fix now or when we move to the Python API - per a conversation with Rob earlier.

I will modify this to be in line with what the schema subcommand is doing - i.e. throw an AttributeError exception that is caught in commands.py.